Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcFileCopier.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Hashing;
+using BuildXL.Cache.ContentStore.Interfaces.FileSystem;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+using BuildXL.Cache.ContentStore.Interfaces.Utils;
+using BuildXL.Cache.ContentStore.Service.Grpc;
+
+namespace BuildXL.Cache.ContentStore.Distributed.Utilities
+{
+    /// <summary>
+    /// File copier which operates over Grpc. <seealso cref="GrpcCopyClient"/>, <seealso cref="GrpcServerFactory"/>
+    /// </summary>
+    public class GrpcFileCopier : IAbsolutePathFileCopier
+    {
+        private Context _context;
+        private int _grpcPort;
+
+        /// <summary>
+        /// Constructor for <see cref="GrpcFileCopier"/>.
+        /// </summary>
+        public GrpcFileCopier(Context context, int grpcPort)
+        {
+            _context = context;
+            _grpcPort = grpcPort;
+        }
+
+        /// <inheritdoc />
+        public Task<FileExistenceResult> CheckFileExistsAsync(AbsolutePath path, TimeSpan timeout, CancellationToken cancellationToken)
+        {
+            // TODO: Implement!
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc />
+        public async Task<CopyFileResult> CopyFileAsync(AbsolutePath sourcePath, AbsolutePath destinationPath, long contentSize, bool overwrite, CancellationToken cancellationToken)

```

Comment:
>CopyFileAsync [](start = 42, length = 13)

It seems that there is an assumptoin that this path is from the cache folder (i.e. extension `.blob`) should we document or validate this?